### ISSUE TITLE
Any build-in Metric class for accuracy of the  CrfTagger?

### ISSUE THREAD

**Issue by XinliYu** [allenai/allennlp#3087]
The CrfTagger class' forward function outputs `tags` as its label prediction (https://github.com/allenai/allennlp/blob/master/allennlp/models/crf_tagger.py#L177-L178). Is there any build-in Metric class can be used to measure its accuracy? 

The CategoricalAccuracy seems to only accept logits and the true label tensor as the inputs, not the predicted labels and true labels (https://github.com/allenai/allennlp/blob/master/allennlp/training/metrics/categorical_accuracy.py#L36-L40). 

CrfTagger does output the `logits`. I am not sure but I think the `logits` output of the CrfTagger are the scores before CRF, and cannot be used to measure the accuracy?

**Comment by matt-gardner** [comment]
These are the metrics that the CRF tagger uses: https://github.com/allenai/allennlp/blob/417a75727d4604e1da0eda461b03b29a73f0bf50/allennlp/models/crf_tagger.py#L123-L134

The metrics that we compute are correct.  A softmax over the logits does not impact the relative ordering of the labels, only the absolute value, so passing the logits to `CategoricalAccuracy` is sufficient.

But it turns out that the CRF further does viterbi decoding before sending fake "class probabilities" to the metrics: https://github.com/allenai/allennlp/blob/417a75727d4604e1da0eda461b03b29a73f0bf50/allennlp/models/crf_tagger.py#L209-L214
